src/authhelper.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+  authhelper.cpp
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+  Copyright (C) 2016-2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <KAuth>
+#include <QCoreApplication>
+#include <QFile>
+#include <QObject>
+#include <QProcess>
+
+class AuthHelper : public QObject
+{
+    Q_OBJECT
+public:
+    AuthHelper() = default;
+
+public slots:
+    KAuth::ActionReply elevate(const QVariantMap& args)
+    {
+        // make install installs the helper to /usr/lib/kauth/ and the elevate script to /usr/lib/libexec
+        const QString script = qApp->applicationDirPath() + (QStringLiteral("/../libexec/elevate_perf_privileges.sh"));
+
+        if (!QFile::exists(script)) {
+            return KAuth::ActionReply::HelperErrorReply();
+        }
+
+        QProcess process;
+        process.start(script, {args[QStringLiteral("output")].toString()});
+        process.waitForFinished(1000);
```

Comment:
check the return value for this call - even if you say "it cannot fail". it could be that someone edits the script, making it fail. or $whatever - no need to assume success when it's so easy to check it here!

Reply:
I added a check that reports an error if the script is not stopped by waitForFinished.

Reply:
did you forget to push? I can't see it?

Reply:
oh now I see it - the connect above. that's far too complicated - just use the return value of `waitForFinished()!`

```
if (process.waitForFinished(1000))
    return KAuth::ActionReply::SuccessReply();
// can we include/show the `process.errorString()`?
return KAuth::ActionReply::HelperErrorReply();

Reply:
meh too easy